timeseries_storage/blueflood/blueflood.go

```diff
@@ -450,6 +450,44 @@ var samplerMap map[api.SampleMethod]sampler = map[api.SampleMethod]sampler{
 	},
 }
 
+// Blueflood will use the finest-grained resolution which doesn't exceed the slot limit.
+// Thus, if you request too many points, it will automatically downsample the data.
+func (b *Blueflood) AdjustTimerange(requested api.Timerange, slotLimit int) api.Timerange {
+	// In some cases, coarser-resolution data may have a shorter TTL.
+	// To accomodate these cases, it must be verified that the requested timerange will
+	// actually be present for the chosen resolution.
+	// TODO: figure out how to make this work with moving averages and timeshifts
```

Comment:
This will also interact with my PR (#204) since we're doing time range altering there as well. Maybe we need to "adjust" the timerange before any functions are applied?